xpdtools/tools.py

```diff
@@ -581,3 +584,89 @@ def inner(x, *args, **kwargs):
         return func(*args, **kwargs)
 
     return inner
+
+
+
+def find_sample_from_2dscan(I_arr, xy_arr, Q_arr=None, 
```

Comment:
I am a little confused about whether this is in the right place.  It seems that most things in ``xpdtools.tools`` are fairly low-level functions for the functioning of the pipelines.  On the other hand, I think this is some kind of script for actually doing xy scans to find a sample?  Or is it for analyzing 2d data from such a scan to find positions of the sample given the data?  I am afraid that if we don't hae the right structure to our files and projects it will become a maintenance (and user) nightmare.  Can we have a discussion (maybe a call) to discuss some of these issues?  I would like to define different activities and group code accordingly.